src/Microsoft.AspNet.Mvc.ModelBinding/Internal/ModelMetadataAttributeHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Internal
+{
+    internal class ModelMetadataAttributeHelper
+    {
+        internal IEnumerable<Attribute> GetModelMetadataAttributesOnType(Type type, IEnumerable<Attribute> attributes)
+        {
+            var modelMedatadataType = type.GetTypeInfo().GetCustomAttribute<ModelMetadataTypeAttribute>();
+            if (modelMedatadataType != null)
+            {
+                var modelMedatadataAttributes = modelMedatadataType.MetadataBuddyType.GetTypeInfo().GetCustomAttributes();
+
+                //If same attribute exists in both attributes and modelMedatadataAttributes collection, 
+                //pick from attributes collection(view model)
+                attributes = attributes.Concat(modelMedatadataAttributes).GroupBy(a => a.GetType())
+                          .Select(grp => grp.First());
+            }
+
+            return attributes;
+        }
+
+        internal IEnumerable<object> GetModelMetadataAttributesOnProperty(Type type, IEnumerable<object> attributes, string propertyName)
```

Comment:
I see part of this: one of the two users of these methods (`AssociatedMetadataProvider`) needs additional attributes.  could the other user (`AssociatedValidatorProvider`) get the same attribute collection w/o harm?  that would at least avoid this wrapping.